probe/core/src/main/java/org/jboss/weld/probe/Reports.java

```diff
@@ -0,0 +1,369 @@
+import java.nio.file.Files;
+import java.security.AccessController;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.AmbiguousResolutionException;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.jboss.weld.Container;
+import org.jboss.weld.bootstrap.api.Environment;
+import org.jboss.weld.bootstrap.spi.BeanDeploymentArchive;
+import org.jboss.weld.bootstrap.spi.BeanDiscoveryMode;
+import org.jboss.weld.bootstrap.spi.BeansXml;
+import org.jboss.weld.config.ConfigurationKey;
+import org.jboss.weld.config.Description;
+import org.jboss.weld.config.WeldConfiguration;
+import org.jboss.weld.manager.BeanManagerImpl;
+import org.jboss.weld.probe.Components.BeanKind;
+import org.jboss.weld.security.GetSystemPropertyAction;
+import org.jboss.weld.util.reflection.Formats;
+
+/**
+ *
+ * @author Martin Kouba
+ */
+class Reports {
+
+    private static final String VALIDATION_REPORT_FILE_NAME = "weld-validation-report.html";
+
+    private static final String TITLE = "Weld - Validation Report";
+    private static final String TITLE_EXCEPTION = "Validation Issue";
+    private static final String TITLE_BDAS = "Deployment - Bean Archives";
+    private static final String TITLE_DEPS = "Dependency Issues";
+    private static final String TITLE_CONFIG = "Weld Configuration - Modified Values";
+    private static final String TITLE_BEANS = "Enabled Beans";
+
+    private static final String EXCEPTION = "exception";
+    private static final String BDAS = "bdas";
+    private static final String DEPS = "deps";
+    private static final String CONFIG = "config";
+    private static final String BEANS = "beans";
+
+    private Reports() {
+    }
+
+    static void exportValidationReport(Probe probe, Exception exception, Environment environment, BeanManagerImpl manager) {
+
+        HtmlTag html = HtmlTag.html();
+        HtmlTag head = html.addAndGet(HtmlTag.head());
+        head.add(HtmlTag.title(TITLE));
+        head.add(HtmlTag.style().add(IOUtils.getResourceAsString("/report.css")));
+
+        HtmlTag body = html.addAndGet(HtmlTag.body());
+
+        body.add(HtmlTag.h1(TITLE));
+        HtmlTag meta = body.addAndGet(HtmlTag.stripedTable());
+        meta.add(HtmlTag.tr().add(HtmlTag.td().add(HtmlTag.strong("Generated at:")), HtmlTag.td(LocalDateTime.now().toString())));
+        meta.add(HtmlTag.tr().add(HtmlTag.td().add(HtmlTag.strong("Weld Version:")), HtmlTag.td(Formats.getSimpleVersion())));
+        meta.add(HtmlTag.tr().add(HtmlTag.td().add(HtmlTag.strong("Weld Environment:")), HtmlTag.td(environment.toString())));
+        meta.add(HtmlTag.tr().add(HtmlTag.td().add(HtmlTag.strong("Java Version:")),
```

Comment:
Just minor suggestion - displaying `java.vendor` could probably be helpful as well.